src/mmrelay/meshtastic_utils.py

```
@@ -1175,12 +1458,89 @@ def connect_meshtastic(
+                                # still running (often due to a hung BlueZ/DBus call), we skip queuing
+                                # a new task. Raising TimeoutError here intentionally reuses the
+                                # existing retry/backoff logic rather than silently proceeding.
+                                with _ble_executor_lock:
+                                    if _ble_future and not _ble_future.done():
+                                        logger.debug(
+                                            "BLE worker busy; skipping interface creation for %s",
+                                            ble_address,
+                                        )
+                                        raise TimeoutError(
+                                            f"BLE interface creation already in progress for {ble_address}."
+                                        )
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)